modules/sdk/project-templates/project-templates/src/test/java/com/liferay/project/templates/ProjectTemplatesTest.java

```diff
@@ -3153,6 +3186,66 @@ private void _testBuildTemplateWithWorkspace(
 		_testExists(workspaceProjectDir, jarFilePath);
 	}
 
+	private void _testPomXmlContainsDependency(
+			Path pomXmlPath, String groupId, String artifactId, String version)
+		throws Exception {
+
+		DocumentBuilderFactory documentBuilderFactory =
+			DocumentBuilderFactory.newInstance();
+
+		DocumentBuilder documentBuilder =
+			documentBuilderFactory.newDocumentBuilder();
+
+		Assert.assertTrue("Missing " + pomXmlPath, Files.exists(pomXmlPath));
+
+		Document document = documentBuilder.parse(pomXmlPath.toFile());
+
+		Element projectElement = document.getDocumentElement();
+
+		Element dependenciesElement = XMLTestUtil.getChildElement(
+			projectElement, "dependencies");
+
+		List<Element> dependencyElements;
+
+		if (dependenciesElement != null) {
+			dependencyElements = XMLTestUtil.getChildElements(
+				dependenciesElement);
+		}
+		else {
+			dependencyElements = Collections.emptyList();
+		}
+
+		boolean foundDependency = false;
+
+		for (Element dependencyElement : dependencyElements) {
+			String dependencyElementString = XMLTestUtil.toString(
+				dependencyElement);
+
+			String artifactIdString = String.format(
+				"<artifactId>%s</artifactId>", artifactId);
+			String groupIdString = String.format(
+				"<groupId>%s</groupId>", groupId);
+			String versionString = String.format(
+				"<version>%s</version>", version);
+
+			if (dependencyElementString.contains(artifactIdString) &&
+				dependencyElementString.contains(groupIdString) &&
+				dependencyElementString.contains(versionString)) {
+
+				foundDependency = true;
+
+				break;
+			}
+		}
+
+		String dependencyString = groupId + ":" + artifactId + ":" + version;
+
+		String missingDependencyString = String.format(
+			"Missing dependency %s in %s", dependencyString, pomXmlPath);
```

Comment:
Since you're already using `String.format`, why don't you pass `groupId`, `artifactId` and `version` as arguments, instead of creating a temporary string?

Reply:
Good point! I will update it :)